### PR TITLE
refactor: BEM convention for common components

### PR DESCRIPTION
**Pull request by ahtesham-quraish**
Description:

1. Implemented BEM convention for common components
2. Moved the classes which were related to common components only few lefts which were used in some other components

How Has This Been Tested?
it's been tested locally.

Its related to: https://2u-internal.atlassian.net/browse/VAN-1425

### CODE REVIEW

File: src/recommendations/RecommendationsList.jsx [hunk 1]
```diff
@@ -10,10 +10,10 @@ const RecommendationsList = (props) => {
 
   return (
     <Container id="course-recommendations" size="lg" className="recommendations-container">
-      <h2 className="text-sm-center mb-4 text-left recommendations-heading">
+      <h2 className="text-sm-center mb-4 text-left recommendations-container__recommendations-heading">
```
Comment (syedsajjadkazmii) [inline src/recommendations/RecommendationsList.jsx:13]:
```suggestion
      <h2 className="text-sm-center mb-4 text-left recommendations-container__heading">
```

File: src/sass/_login_component.scss [hunk 2]
```diff
@@ -0,0 +1,3 @@
+.login-button-width {
+    min-width: 6rem;
+}
```
Comment (syedsajjadkazmii) [inline src/sass/_login_component.scss:3]: please add EOL

File: src/sass/_progressive_profiling_component.scss [hunk 3]
```diff
@@ -0,0 +1,26 @@
+.pp-page--login-button-width{
```
Comment (syedsajjadkazmii) [inline src/sass/_progressive_profiling_component.scss:1]: can we rename the file to `_progressive_profiling_page.scss`? This suggestion applies to login scss file as well.

File: src/common-components/InstitutionLogistration.jsx [hunk 4]
```diff
@@ -42,7 +42,7 @@ const InstitutionLogistration = props => {
     <>
       <div className="d-flex justify-content-left mb-4 mt-2">
         <div className="flex-column">
-          <h4 className="mb-2 font-weight-bold institute-heading">
+          <h4 className="mb-2 font-weight-bold institute--heading">
```
Comment (syedsajjadkazmii) [inline src/common-components/InstitutionLogistration.jsx:45]: should not it be? 
```suggestion
          <h4 className="mb-2 font-weight-bold institutions__heading">
```

File: src/common-components/InstitutionLogistration.jsx [hunk 5]
```diff
@@ -57,7 +57,7 @@ const InstitutionLogistration = props => {
               <tr key={provider} className="pgn__data-table-row">
                 <td>
                   <Hyperlink
-                    className="btn nav-item p-0 mb-1 secondary-provider-link"
+                    className="btn nav-item p-0 mb-1 institute--secondary-provider-link"
```
Comment (syedsajjadkazmii) [inline src/common-components/InstitutionLogistration.jsx:60]:
```suggestion
                    className="btn nav-item p-0 mb-1 institutions--provider-link"
```

File: src/progressive-profiling/ProgressiveProfiling.jsx [hunk 6]
```diff
@@ -177,7 +177,7 @@ const ProgressiveProfiling = (props) => {
       ) : null}
       <div className="mw-xs m-4 pp-page-content">
         <div>
-          <h2 className="pp-page-heading text-primary">{formatMessage(messages['progressive.profiling.page.heading'])}</h2>
+          <h2 className="pp-page--heading text-primary">{formatMessage(messages['progressive.profiling.page.heading'])}</h2>
```
Comment (syedsajjadkazmii) [inline src/progressive-profiling/ProgressiveProfiling.jsx:180]:
```suggestion
          <h2 className="pp-page__heading text-primary">{formatMessage(messages['progressive.profiling.page.heading'])}</h2>
```

File: src/progressive-profiling/ProgressiveProfiling.jsx [hunk 7]
```diff
@@ -189,7 +189,7 @@ const ProgressiveProfiling = (props) => {
         <Form>
           {formFields}
           {(getConfig().AUTHN_PROGRESSIVE_PROFILING_SUPPORT_LINK) && (
-            <span className="progressive-profiling-support">
+            <span className="pp-page--support">
```
Comment (syedsajjadkazmii) [inline src/progressive-profiling/ProgressiveProfiling.jsx:192]:
```suggestion
            <span className="pp-page__support-link">
```

File: src/progressive-profiling/ProgressiveProfiling.jsx [hunk 8]
```diff
@@ -206,7 +206,7 @@ const ProgressiveProfiling = (props) => {
             <StatefulButton
               type="submit"
               variant="brand"
-              className="login-button-width"
+              className="pp-page--login-button-width"
```
Comment (syedsajjadkazmii) [inline src/progressive-profiling/ProgressiveProfiling.jsx:209]:
```suggestion
              className="pp-page__button-width"
```

Comment (ahtesham-quraish) [reply src/progressive-profiling/ProgressiveProfiling.jsx:209]: yeah i have addressed it .

File: src/progressive-profiling/ProgressiveProfiling.jsx (hunk 7 above)
Comment (ahtesham-quraish) [reply src/progressive-profiling/ProgressiveProfiling.jsx:192]: addressed

File: src/progressive-profiling/ProgressiveProfiling.jsx (hunk 6 above)
Comment (ahtesham-quraish) [reply src/progressive-profiling/ProgressiveProfiling.jsx:180]: addressed

File: src/common-components/InstitutionLogistration.jsx (hunk 4 above)
Comment (ahtesham-quraish) [reply src/common-components/InstitutionLogistration.jsx:45]: we can do it like this

**Review by syedsajjadkazmii — APPROVED** [review]

**Pull request merged**